service/history/replication/task_processor.go

```diff
@@ -465,7 +465,7 @@ func (p *taskProcessorImpl) processTaskOnce(replicationTask *types.ReplicationTa
 	scope, err := p.taskExecutor.execute(replicationTask, false)
 
 	if err != nil {
-		p.updateFailureMetric(scope, err)
+		p.updateFailureMetric(scope, err, p.shard.GetShardID())
```

Comment:
nit: can be simplified

```suggestion
		p.updateFailureMetric(p.metricsClient.Scope(scope, metrics.InstanceTag(strconv.Itoa(p.shard.GetShardID()))), err)
```

Reply:
it will not work since updateFailureMetrics require the int type for the scope